packages/teleport/src/Discover/LoginTrait/LoginTrait.tsx

```diff
@@ -0,0 +1,169 @@
+/**
+ * Copyright 2022 Gravitational, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useState } from 'react';
+import { Flex, ButtonPrimary, ButtonText, Text, Box, Indicator } from 'design';
+import { Danger } from 'design/Alert';
+import * as Icons from 'design/Icon';
+import FieldInput from 'shared/components/FieldInput';
+import Validation from 'shared/components/Validation';
+import { Header, CancelButton } from '../Shared';
+import type { AgentStepProps } from '../Shared';
+import { useDiscoverContext } from '../discoverContextProvider';
+import { useLoginTrait, State } from './useLoginTrait';
+
+export default function Container(props: AgentStepProps) {
+  const ctx = useDiscoverContext();
+  const state = useLoginTrait({ ctx, props });
+
+  return <LoginTrait {...state} />;
+}
+
+export function LoginTrait({
+  attempt,
+  nextStep,
+  loginMap,
+  toggleLoginSelect,
+  addLogin,
+}: State) {
+  const [newLogin, setNewLogin] = useState('');
+  const [showInputBox, setShowInputBox] = useState(false);
+
+  return (
+    <Box>
+      <Header>Set Up Access</Header>
+      {attempt.status === 'processing' && (
+        <Box textAlign="center" m={10}>
+          <Indicator />
+        </Box>
+      )}
+      {attempt.status === 'failed' && <Danger>{attempt.statusText}</Danger>}
+      {attempt.status === 'success' && (
```

Comment:
also there is also another way to express that using object literals
```
const content = {
   processing: <Box textAlign="center">...etc</Box>,
   failed: ... ,
   ...
}[attempt.status]
```

This compact syntax comes with slight perf hit since all "branches" needs to allocate.

Reply:
Using flags for hidding/showing components is a way to keep JSX scoped in
one place in regards to layout. This also makes it more readable as it reduces 
the number of "jumps" to see what supposed to be rendered (bigger picture). 
It allows people to read JSX from up to bottom in one flow as it does not break the scope (more declarative). 

Essentially this is a shortcut for this:

```
<Box visible={true}>
  <Content/>
</Box>
``` 

which becomes

```
{ flag &&  (  <Content/> )}
```

and which is better than just: 

```
 {$content} // provides no information about what and how supposed to be rendered
```

One thing to keep in mind is to avoid complex expressions and nested conditions. Usually it means refactoring the code to a separate component by declaratively passing props to it.